moved all settings to platform_interface for simpler crossplatform integration; added `recoverImage` method to get the cropped image in case MainActivity was terminated during cropping

yendacoder opened the pull request:
See detailed comment in for the `recoverImage()` method in `cropper.dart`

hnvn commented:
Thanks for your PR. This address an issue that I am working on, the complexity of integration if you have to support Web, my temporary solution is to use conditional import but it leads to complex and verbose.  

 My comment: your PR changes settings models, so I think you need to update README document too.

yendacoder commented:
Hi. Sorry for the mess. 

Initially, the PR was intended to solve the issue with lost crop results on Android due to host Activity termination. Since you didn't respond to it, I got desperate and published my fork under a different name since I need the change in my project. Today made changes for simpler web integration and accidentally pushed all to the same branch. So, it was all over the place, together with renamed files and everything.

I think I've reverted the files I've renamed and also updated the readme (basically, just dropped the part that is not relevant anymore), but it surely needs a double check.

I'd be happy if all my changes make it to the original repo, but it wasn't supposed to be this messy.

hnvn commented:
LGTM
